borg/helpers.py

```diff
@@ -722,7 +732,7 @@ def resize(self, size, init=False):
         """
         size = int(size)
         if self.limit is not None and size > self.limit:
-            raise ValueError('Requested buffer size %d is above the limit of %d.' % (size, self.limit))
+            raise Buffer.OutOfMemory(size, self.limit)
```

Comment:
is this enough to initialize the OSError, so accessing .errno, .path, .strerror works? (we access them when transforming OSError to our special error class for input file os errors).